Core/DetectorConstruction.cpp

```diff
@@ -238,6 +238,25 @@ void DetectorConstruction::InitializeMaterials()
     G4Material::GetMaterial("G4_NYLON-6-6")
   );
   fSmallChamberRun7Material->AllowsAnnihilations(true);
+  
+  
+  //! Vacuum
+  G4double z, a;
+  G4String name;
+  G4double density = 1.e-25 *g/cm3;
```

Comment:
Density should be linear to the pressure (p = d *C -> p/2 = d/2 * C) , so If we linearly extrapolate the pressure and density of the air from point 
101325 Pa : 1.225 g/cm3
to 
10^(-19),
it will give a mass of 1.209 E-24 g/cm3
What do you think about changing the code as follows (this will allow for user to change the pressure in the chamber from the macro):
- Inserting in DetectorConstruction.h (after the line 162 with   G4int maxScinID = 512;) line with definition of the pressure:
G4double fPressure = 1.e-19 *pascal;
- Removing line with pressure in here (line 248 DetectorConstruction.cpp)
- Changing here in line 253 pressure to fPressure
- Changing here line 246 to:
G4double density = (fPressure/1.e-19 *pascal) * 1.209e-24*g/cm3;
and additional comment after: //Density linearly extrapolated from the air density and pressure
- Inserting in DetectorConstruction.h (after the line 70 with   void ReloadMaterials(const G4String& material);) new function:
void SetPressureInChamber(G4double pressure) {fPressure = pressure;};
- Creating in DetectorConstructionMessenger.h new pointer (after the line 52 with  G4UIcmdWithAString* fCreateGeometryType = nullptr;):
G4UIcmdWithADoubleAndUnit* fPressureInChamber = nullptr;
- Inserting in DetectorConstructionMessenger.cpp (after in line 101 with    }):
} else if (command == fPressureInChamber) {
    fDetector->SetPressureInChamber(newValue)
}
- Inserting in DetectorConstructionMessenger.cpp (after the line 65 with   delete fCreateGeometryType;):
delete fPressureInChamber;
- Inserting in DetectorConstructionMessenger.cpp (after the line 53 with   fCreateGeometryType->SetGuidance("Set structure of output JSON file: barrel or modular.");):
  fPressureInChamber = new G4UIcmdWithADoubleAndUnit("/jpetmc/detector/chamberPressure", this);
  fPressureInChamber->SetGuidance("Define pressure in the chamber");
  fPressureInChamber->SetDefaultUnit("Pa");
  fPressureInChamber->SetUnitCandidates("Pa");

Reply:
This is a good idea. Thanks!